tests/client/amo/test_utils.js

```diff
@@ -87,5 +92,35 @@ describe('amo/utils', () => {
           mockApi.verify();
         });
     });
+
+    it('returns a rejected Promise if the addonsType is wrong', () => {
+      ownProps.params.visibleAddonType = '404';
+      const store = createStore({ application: 'android' });
+
+      return loadLandingAddons({ store, params: ownProps.params })
```

Comment:
I know that the `beforeEach` function protects it, but mutating objects like this is risky and I think an approach with cloning reads just as well:
````js
return loadLandingAddons({ 
  store, 
  params: { ...ownProps.params, visibleAddonType: '404' }, 
})
  .then(...);
````